### PR TITLE
John conroy/detail header suggestions

### LINKED ISSUE #1246: Rename `collectionsDivider` inside theme palette.
Now that we're consuming this color in other components the variable should be renamed.

### LINKED ISSUE #1252: Centralize references to header height
References to the header height should come from a single source.

### PR DESCRIPTION
**Pull request by john-conroy**
Implementing suggestions from #1244. Fix #1246, fix #1252, add tests. 

@ilan-gold / @keller-mark there are a few changes to the visualization component.

Screen Shot 2020-09-29 at 10 16 04 AM

### CODE REVIEW

**Review by tsliaw — CHANGES_REQUESTED** [review]
I'm running into error pages for all the donors that have undefined values for everything. Screenshot of error for donor shown below. Error pages are also showing part of the entity header. Examples: [HBM485.KRGM.628](http://localhost:5001/browse/donor/48b15ba8abfaef4e70edfdf173649b6c), [HBM774.MQRL.487](http://localhost:5001/browse/donor/c9a52c518091b9864f81d57ed783b527)
Screen Shot 2020-09-29 at 9 24 57 AM

**Comment by john-conroy** [comment]
As we discussed on slack, the test env has odd data and we expect donor mapped metadata to exist in some form.

**Review by tsliaw — APPROVED** [review]

**Review by ilan-gold — APPROVED** [review]
Looks good! Thanks!

File: context/app/static/js/components/Detail/entityHeader/EntityHeader/utils.js [hunk 1]
```diff
@@ -1,29 +1,27 @@
-const extractValuesFromObject = (obj, keys) =>
-  keys.reduce((acc, k) => {
-    if (k in obj) {
-      acc.push(obj[k]);
-    }
+const extractAndLabelMetadata = (obj, keys) => {
```
Comment (mccalluc) [inline context/app/static/js/components/Detail/entityHeader/EntityHeader/utils.js:1]: maybe...
```suggestion
function extractAndLabelMetadata(obj, keys) {
```

File: context/app/static/js/components/Detail/entityHeader/EntityHeader/utils.js [hunk 2]
```diff
@@ -1,29 +1,27 @@
...
+  const labels = { mapped_organ: 'organ type', mapped_data_types: 'data type', mapped_specimen_type: 'specimen type' };
+  return keys.reduce((acc, k) => {
```
Comment (mccalluc) [inline context/app/static/js/components/Detail/entityHeader/EntityHeader/utils.js:3]:
```
const acc = {};
keys.foreach((k) => {
  acc[k] = { value: k in obj ? obj[k] : undefined, label: k in labels ? labels[k] : k };
});
return acc;
```
I'm not sure that `reduce` is the clearest way to explain what's going on here... but not a strongly held opinion.

File: context/app/static/js/components/Detail/entityHeader/EntityHeaderContent/EntityHeaderContent.jsx [hunk 3]
```diff
@@ -26,7 +26,9 @@ function EntityHeaderContent({ display_doi, entity_type, data, summaryInView })
         <AnimatedFlexContainer style={props} key={key}>
           {iconMap[entity_type]}
           <EntityHeaderItem text={display_doi} />
-          {data.map((d) => d && <EntityHeaderItem text={d} key={d} />)}
+          {Object.entries(data).map(([k, v]) => (
+            <EntityHeaderItem text={v.value ? v.value : `undefined ${v.label}`} key={k} />
```
Comment (mccalluc) [inline context/app/static/js/components/Detail/entityHeader/EntityHeaderContent/EntityHeaderContent.jsx:30]: - Could`v.value` be zero? (... thought, at least for our current data, probably not)
- Use `||`:
```suggestion
            <EntityHeaderItem text={v.value || `undefined ${v.label}`} key={k} />
```

**Review by mccalluc — APPROVED** [review]
Couple comments, but fine as-is. Thanks!

File: context/app/static/js/components/Detail/entityHeader/EntityHeader/utils.js (hunk 2 above)
Comment (john-conroy) [reply context/app/static/js/components/Detail/entityHeader/EntityHeader/utils.js]: Done.

**Pull request merged**